lib/constants.rb

```diff
@@ -147,7 +147,7 @@ module Constants
     "sncf_tvs_id"          => '[A-Z]{3}(_TRANS|_PSL)?',                 # ABC, ABC_TRANS, ABC_PSL
     "trenitalia_id"        => '[0-9]{7}',                               # 1234567
     "trenitalia_rtvt_id"   => '[NS][0-9]{5}',                           # N12345, S12345
-    "uic"                  => '[0-9]{7}',                               # 1234567
+    "uic"                  => '[0-9]{7,8}',                             # 1234567
```

Comment:
UIC codes are always with 7 numbers. SNCF have their own extension for UIC codes on 8 numbers which we've dubbed `uic8_sncf`. You can obtain the 7 number version by dropping the last digit.